### PR TITLE
Layout Grid Availability -  featureConfig implementation

### PR DESCRIPTION
**Pull request by enejb**
This PR is similar to https://github.com/wordpress-mobile/WordPress-Android/pull/13804

It adds the availability information that is needed for the layout grid block to be available to only .com sites for now and in the future to all sites that have the layout grid block.

Related PR:
https://github.com/WordPress/gutenberg/pull/31777
Related https://github.com/wordpress-mobile/gutenberg-mobile/pull/3513

To test:

CASE A: 
    - An business site as well as a simple site should see the layout grid block.

CASE B: 
     - An Jetpack site or a .org site should NOT see the layout grid block.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 1]
```diff
@@ -353,6 +353,10 @@ public static boolean supportsContactInfoFeature(SiteModel site) {
         return site != null & (site.isWPCom() || checkMinimalJetpackVersion(site, WP_CONTACT_INFO_JETPACK_VERSION));
     }
 
+    public static boolean supportsLayoutGridFeature(SiteModel site) {
+        return site != null & (site.isWPCom() || site.isWPComAtomic());
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java]: The use of single `&` in the two cases above should be a typo 🤔

**Review by antonis — APPROVED** [review]
Great work @enejb 👍 
The code changes LGTM and the app works as expected 🎉 
I tested this both with metro and the CI produced [APK](https://106742-9306568-gh.circle-artifacts.com/0/Artifacts/WordPress-pr-14667-build-106742.apk):
* I took the liberty of pushing a version change pointing at the[ latest commit that has an artifact on S3](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3513/commits/ba6208281a9129514b76b0093609e1b7b7c8ac16) to produce the APK.
* The remaining failing lint check is due to a missing line at the end of `LayoutGridBlockFeatureConfig.kt`

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 2]
```diff
@@ -350,7 +350,11 @@ public static boolean supportsStoriesFeature(SiteModel site) {
     }
 
     public static boolean supportsContactInfoFeature(SiteModel site) {
-        return site != null & (site.isWPCom() || checkMinimalJetpackVersion(site, WP_CONTACT_INFO_JETPACK_VERSION));
+        return site != null && (site.isWPCom() || checkMinimalJetpackVersion(site, WP_CONTACT_INFO_JETPACK_VERSION));
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:353]: thanks for this fix 🙇

File: build.gradle [hunk 3]
```diff
@@ -7,7 +7,7 @@ buildscript {
     ext.kotlin_ktx_version = '1.2.0'
     ext.wordPressUtilsVersion = '1.40.0'
     ext.detektVersion = '1.15.0'
-    ext.gutenbergMobileVersion = 'v1.53.0-alpha1'
+    ext.gutenbergMobileVersion = '3513-ba6208281a9129514b76b0093609e1b7b7c8ac16'
```
Comment (mzorz) [inline build.gradle]: question, this would be changed before merging, right?

**Review by mzorz — APPROVED** [review]
Nice work @enejb  - as discussed elsewhere code LGTM, approving but leaving merge for you to do later once conflicts are fixed and the other PRs are all in order to be merged 👍 
🚢

File: build.gradle (hunk 3 above)
Comment (enejb) [reply build.gradle]: Yes we would update it to the correct version that contains the current PRs (3513) - info

**Pull request merged**